### DISCUSSION TITLE
Audiobook status tags and icons

### DISCUSSION THREAD

**Discussion by scorpio862** [advplyr/audiobookshelf#13] [Ideas]
It would be really usefull if all the audiobooks got automatically assigned listening status like: New, Listening, Finished visible on the bookshelf.  It could be a small icon on the bookcover or something like this, so I could clearly see which books are newly added or 100% finished.  It could be used in the filter to show only the books with specific status like only new books, or only the started books. 
I can see that all started books on the bookshelf have a yellow progress bar below the bookcover but this is barely visible and it is dificult to say it the books is 100% finished or almost fininished like 95%.
This is a great project and thank you for your work.

**Comment by advplyr** [comment]
I like this idea. What do you think would be the criteria for "new"?

What do you think the icons should look like?

On another note: I'm thinking of adding an optional feature where the top shelf shows currently listening audiobooks. I often listen to several books at a time, so it would be convenient for me to have those easily accessible. Either that or creating another view with each shelf being something like listening, new, etc. like netflix.

**Comment by scorpio862** [comment]
By "new" audiobooks, I'm thinking about the ones that where discovered by the latest scan.
I'm planing to build a large library with audiobooks, so when I'm adding new books to existing library it will be easier to find them on the bookshelf. 

When I started thinking about the style of the icons I noticed that it is not a good idea to put any icons on the book cover as this potentially could cover up the name of the book. 
So the icons need to be subtle.

New books could be marked like this

![image](https://user-images.githubusercontent.com/50056988/131224268-e9eef5a7-822a-44b1-899e-52670dc9a339.png)

It would good if the Finished audiobooks where marked somehow. Maybe by changing the orange progress bar to a different colour when book progress is at 100%

![image](https://user-images.githubusercontent.com/50056988/131224646-4cd6aa17-db4d-4da2-90ea-56aad47faa3b.png)



Great idea with the top shelf showing currently listening audiobooks. Probably the same could be done with new books.

**Comment by Merijeek** [comment]
I'd suggest that "new" just means "not begun" and not finished. 

Otherwise, it'll discourage people scanning for new. That and it'd likely fast become meaningless when you implement the inevitable auto-add function that adds books as they are added to the watched folder. 

And I agree that having a top shelf that shows currently in progress would be a good idea. 

A suggestion for your app would be to allow local storage that then syncs when reconnected. I'd prefer to not forward ports on a firewall, and sometimes access to the home network via VPN isn't convenient.

**Comment by scorpio862** [comment]
New audiobooks could be the ones that has never been started and finished but I think it would be good to have an option to unmark them as new. 
For example if I added 100 audiobooks and never listened to most of them, I don't want them to be still marked as new after a year.

**Comment by Merijeek** [comment]
Oh, most certainly there should always be a way to manually mark as new or old. No doubt there.

**Comment by advplyr** [comment]
I like that new flag design you have there, will definitely use that. I don't think every audiobook should be flagged as new on the first scan. Any books added after the initial scan will get the new flag.

If you have anymore design ideas I could really use some help in that department. For one, I want to find a new icon/logo, the one used now was not meant to be permanent. The bookshelf and covers look good, but the other colors and buttons don't, I just don't have an eye for design so any suggestions are appreciated.

**Comment by advplyr** [comment]
> A suggestion for your app would be to allow local storage that then syncs when reconnected. I'd prefer to not forward ports on a firewall, and sometimes access to the home network via VPN isn't convenient.

If you haven't forwarded ports how are you accessing the Web GUI from outside your home network?

**Comment by scorpio862** [comment]
No problem. I could help out with the designs. 
What do you think about this?

![image](https://user-images.githubusercontent.com/50056988/131234474-f906fafd-d685-465f-bbe6-71727b774fac.png)

**Reply by advplyr** [reply]
That's great! Very similar to what I was thinking. What version of that icon would we use in the upper-left?

I really appreciate the help and suggestions, this seemed like a great opportunity to try out the new github discussion board.

**Reply by scorpio862** [reply]
What other versions of the icon are you thinking about? Without the box?
The version without the background doesn't look good for me. I like this icons with the background
I looks better with white stroke and I made few options to choose from.

![image](https://user-images.githubusercontent.com/50056988/131265177-46eab946-7132-4031-9a5b-05ec28b0dac3.png)

**Reply by scorpio862** [reply]
Let me know which version do you like, so I can export it and send it to you.

**Comment by advplyr** [comment]
![image](https://user-images.githubusercontent.com/67830747/131234694-d61c98f8-9d83-4958-8fbe-cc0e08dc4821.png)

I'm thinking that "New" and "Finished" could be tags that are automatically set and removed. This way they can be manually removed as easily as editing any other tag.
What do you think of having the "Finished" line on the bottom be the same color green as the new marker is?

**Reply by scorpio862** [reply]
Good idea with status New and Finished as Tags. 
I think the progress bar changing to green should look good.

**Comment by Merijeek** [comment]
Personally, I like the icon without the background the most, but obviously that's mostly just a personal preference on the aesthetic. 

As far as the big on the firwall/sync/etc, what I was envisioning is this: 

Playing via the web on my PC. But now I want to go somewhere, so on the phone app hit the download button. OK, so now I leave, I listen, eventually I return home and am back on my local wifi. Now the app can reach the server again. So progress on any given track gets synced to the server. 

Maybe I want to continue on the phone. Maybe I want to continue on the PC. Whatever, they're synced and can stay that way until I leave again. 

Just a thought.

**Reply by advplyr** [reply]
Oh okay, got it. That shouldn't be much of a problem to implement. I'm trying to put out the app as soon as it is functional, so I'm not sure if a download option will be on the initial beta version. The audio player is looking great so far streaming from the server on android.
